bodo/tests/test_df_lib/test_end_to_end.py

```diff
@@ -682,34 +686,36 @@ def test_head(datapath):
 
 def test_apply(datapath, index_val):
     """Very simple test for df.apply() for sanity checking."""
-    df = pd.DataFrame(
-        {
-            "a": pd.array([1, 2, 3] * 10, "Int64"),
-            "b": pd.array([4, 5, 6] * 10, "Int64"),
-            "c": ["a", "b", "c"] * 10,
-        },
-        index=index_val[:30],
-    )
-    bdf = bd.from_pandas(df)
-    out_pd = df.apply(lambda x: x["a"] + 1, axis=1)
-    out_bodo = bdf.apply(lambda x: x["a"] + 1, axis=1)
+    with assert_executed_plan_count(1):
+        df = pd.DataFrame(
+            {
+                "a": pd.array([1, 2, 3] * 10, "Int64"),
+                "b": pd.array([4, 5, 6] * 10, "Int64"),
+                "c": ["a", "b", "c"] * 10,
+            },
+            index=index_val[:30],
+        )
+        bdf = bd.from_pandas(df)
+        out_pd = df.apply(lambda x: x["a"] + 1, axis=1)
+        out_bodo = bdf.apply(lambda x: x["a"] + 1, axis=1)
```

Comment:
The execution occurs in the apply frontend but the output is still lazy?  So , perhaps add a lazy check for out_bodo here.